cloudbuild.yaml

```diff
@@ -0,0 +1,40 @@
+
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+steps:
+  #
+  # Creates the initial make + docker build platform
+  #
+  - name: ubuntu
+    script: |
+      echo 'FROM gcr.io/cloud-builders/docker\nRUN apt-get install make\nENTRYPOINT [\"/usr/bin/make\"]' > Dockerfile.build
+  - name: gcr.io/cloud-builders/docker
+    id: build-make-docker
+    entrypoint: docker
+    args: [build, -f, Dockerfile.build, -t, make-docker, .]
+
+  # build and push crd-client image to GCR
+  - name: make-docker
+    id: push
+    dir: .
+    env: ['REPOSITORY=${_REPOSITORY}']
+    script: |
+      make push
+options:
+  dynamic_substitutions: true
+  defaultLogsBucketBehavior: REGIONAL_USER_OWNED_BUCKET
+substitutions:
+  _REPOSITORY: YOUR_REPOSITORY_HERE
```

Comment:
I _think_ if you just remove the `_REPOSITORY` substition, the customer would have to provide the substitution on the command line, which would make this file generic.

Reply:
ohhhh that's better indeed